content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,423 @@
+```
+
+Before we go any further, this is a great point to add these changes to your Git
+history. You can do that with the following commands:
+
+```bash
+git add .
+git commit -m "created DVC pipeline"
+```
+
+## Enabling DVC pipelines for checkpoint experiments
+
+After running the command above to setup your _train_ stage, your _dvc.yaml_
+should have the following code.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory. This will create a new stage with checkpoints enabled, but
+does not update existing stages.
+
+The rest of the command `dvc stage add` sets up our dependencies for running the
+training code, which parameters we want to track (which are defined in the
+_params.yaml_), some configurations for our plots and showing the training
+metrics.
+
+We'll manually add a slight modification to the way our plots are handled. Add
+the following lines to your _dvc.yaml_. This just changes the way our results
+are shown after we've run some experiments.
+
+```yaml
+plots:
+  - predictions.json:
+      cache: false
+      template: confusion
+      x: actual
+      y: predicted
+```
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
+
+This is where DVC will be able to track our metrics over time and where we will
+add our checkpoints to give us the points in time with our model that we can
+switch between.
+
+Now we need to enable checkpoints at the code level. We are interested in
+tracking the metrics along with each checkpoint, so we'll need to add a few
+lines of code.
+
+In the `train.py` file, import the `dvclive` package with the other imports:
+
+```bash
+import dvclive
+```
+
+Then update the following lines of code in the `main` method inside of the
+training epoch loop.
+
+```python
+for k, v in metrics.items():
+    print('Epoch %s: %s=%s'%(i, k, v))
+    dvclive.log(k, v)
+dvclive.next_step()
+```
+
+By adding the two `dvclive` methods to our training epoch loop, we are enabling
+checkpoints in our code and recording the training metrics at each of those
+checkpoints.
```

Comment:
A bit more details can be helpful: "To be more specific, `dvclive.log(k, v)` stores metric `k` with a value `v` in plain text files (in `dvclive/` directory by default) and `dvclive.next_step()` signals DVC that it can take a snapshot of the entire workspace and version it with Git. It is important the with this approach only code with meta information is versioned in Git (as an ephemeral commit) while the actual model weight file will be stored in DVC data cache."